### PR TITLE
Improve typechecking, docs, and minor bugs with Field/FieldInfo

### PR DESCRIPTION
**Pull request by dmontagu**
Selected Reviewer: @lig

### CODE REVIEW

File: tests/test_aliases.py [hunk 1]
```diff
@@ -424,14 +424,13 @@ class Model(BaseModel):
...
         (AliasChoices('a', AliasPath('b', 1)), [['a'], ['b', 1]]),
-        (AliasChoices(), None),
     ],
 )
 def test_validation_alias_path(input, expected):
     class Model(BaseModel):
         x: str = Field(validation_alias=input)
 
-    assert Model.model_fields['x'].validation_alias == expected
+    assert Model.model_fields['x'].validation_alias == input
```
Comment (hramezani) [inline tests/test_aliases.py:433]: if we don't need `expected`, we can remove it from `parametrize` parameters

**Comment by dmontagu** [comment]
please review

**Comment by dmontagu** [comment]
This adds a change that breaks pydantic-settings. I have opened a PR that will fix that in https://github.com/pydantic/pydantic-settings/pull/59. I will disable CI for pydantic settings here for now then we can reenable once everything is back in sync.

**Comment by dmontagu** [comment]
Not sure why cloudflare pages is failing but that's the only problem now. (Because I disabled the run of pydantic-settings. We could re-enable once it is updated to use this version of pydantic.)

**Comment by dmontagu** [comment]
Going to try reopening to see if that resolves the cloudflare build issue..

**Comment by dmontagu** [comment]
Hmm nope couldn't get it to work. @samuelcolvin I guess I need some info about the cloudflare thing

**Comment by hramezani** [comment]
@dmontagu Why do we need to add`pydantic-settings` as submodule here?

**Review by hramezani — APPROVED** [review]

**Pull request merged**